### ISSUE TITLE
sameBeanAs(...).with("superclassField", matcher) fails with "superclassField does not exist"

### ISSUE THREAD

**Issue by kozlovda** [shazam/shazamcrest#15]
Hi,

The issue above happens when field is inherited from one of super-classes of the tested bean.

This happens due to shallow lookup of the field at [`BeanFinder.java:33`](https://github.com/shazam/shazamcrest/blob/shazamcrest-0.11/src/main/java/com/shazam/shazamcrest/BeanFinder.java#L33):

```
	private static Object findBeanAt(List<String> fields, Object object) {
		for (Field field : object.getClass().getDeclaredFields()) {
			field.setAccessible(true);
```

As per [`getDeclaredFields()` javadoc](http://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getDeclaredFields()) it doesn't return inherited fields:

> This includes public, protected, default (package) access, and private fields, but excludes inherited fields.

Basically, it seems reasonable to update field lookup to search through inherited fields as well (with all `field shadowing` logic in place).

Thanks in advance for looking into this!
